### PR TITLE
Gzi update GitHub actions

### PR DESCRIPTION
**Pull request by gecage952**
Addressing the last part of the https://code.ornl.gov/sns-hfir-scse/reflectometry/reflectometry/-/issues/376

Specifically, this pr is supposed to add a github workflow to run unit tests and generate a code cov report. Some tests also need to be skipped because of missing data files. Finally, I added some flake rules back that I originally excluded.

### CODE REVIEW

**Comment by gecage952** [comment]
I'm not entirely sure how to test the github actions workflow without just merging it and seeing how it goes. I used other repos under the neutrons group as a reference.

**Comment by gecage952** [comment]
So, I was having issues installing the dependencies (mantid has to be <=6.3, but mantidworkbench wasn't installing since there's no conda package for <6.4 in the nightly channel). I realized though that the unit tests do run even if mantid workbench isn't installed (after running make install though). I'm pretty sure in order to get everything running well, the code just needs to be updated for mantid 6.4. At any rate, the unit tests do run and pass, so it *does* fulfill the reqs of the enabler.

File: setup.cfg [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # configure options here to work for all sub-packages
 [flake8]
-ignore = E401 E402 E722 E741 F821 F841 E203, E266, E501, W503, F403, F401 E712 E713 F811
-max-line-length = 120
+ignore = E402 E722 E741 F821 F841 E203 W503 F401 E712 E713 F811
+max-line-length = 150
```
Comment (peterfpeterson) [inline setup.cfg:4]: Since black (in pre-commit) is forcing lines to be no more than 119 characters, I'm surprised you had to increase this.

Comment (gecage952) [reply setup.cfg:4]: Yeah, these files fail without it: 
Screenshot from 2022-09-20 15-14-34

Comment (gecage952) [reply setup.cfg:4]: Ah, I think they're all comments.

**Review by peterfpeterson — APPROVED** [review]
Given that this isn't aimed at moving to mantid 6.4 (or 6.5 that comes out in a month), I think this was done sufficiently.

**Pull request merged**